core/src/fs/temp_file.cpp

```diff
@@ -6,6 +6,16 @@ USERVER_NAMESPACE_BEGIN
 
 namespace fs {
 
+// Please, review this TempFile::TempFile implementation
+// I simple added it as workaround for linkage error (if linked as shared library)
```

Comment:
Simple: don't use shared linkage with userver

Reply:
Why ? We already debugged shared library variant installation in this request and it works, here much reasons why 99% of libraries on unix systems install in shared variant. And here we try to make this variant available for userver ….

Reply:
There are multiple reasons why we would like to avoid building shared libraries:
* we guarantee no ABI stability even on minor changes. This is usually a deal breaker for shared libraries distributions
* supporting shared libraries would double the testing efforts, double the support efforts, double the build scripts boilerplate and double the development efforts, as it would require `-fvisibility=hidden` and visibility attributes in code
* LTO is not quite nicely supported with shared libraries, we would not like to provoke users for inefficient services